### PR TITLE
feat: [CDS-82820]: Adding validation for Task Manifest having Artifact bundle Store

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Describe your changes
Adding validation for Task Manifest having Artifact bundle Store
here is the tech spec for this : [Tech Spec](https://harness.atlassian.net/wiki/spaces/CDP/pages/21515501604/Tech+Spec+PCF+Fetching+Manifests+from+Artifact+bundle)

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54920)

### CODE REVIEW

**Comment by namantalaycha10** [comment]
changes pending before merge: Make this behind FF

**Comment by namantalaycha10** [comment]
trigger featurenamecheck

**Comment by namantalaycha10** [comment]
trigger uts

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java [hunk 1]
```diff
@@ -935,7 +935,9 @@ public enum FeatureName {
   CDS_REMOVE_RESUME_EVENT_FOR_ASYNC_AND_ASYNCCHAIN_MODE(
       "Remove sdk resume event for async and async chain modes to avoid race condition", PIPELINE),
   CDS_K8S_TRAFFIC_ROUTING_NG(
-      "Providing Traffic Routing functionality for K8s - BG, Canary and TrafficRouting steps", HarnessTeam.CDP);
+      "Providing Traffic Routing functionality for K8s - BG, Canary and TrafficRouting steps", HarnessTeam.CDP),
+  CDS_ENABLE_TAS_ARTIFACT_AS_MANIFEST_SOURCE_NG(
```
Comment (vlica-harness) [inline 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:939]: would it be more sense to rename this to have in the name Artifact Bundle?

Comment (namantalaycha10) [reply 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:939]: @vlica-harness this is what PM approved

**Comment by namantalaycha10** [comment]
trigger uts

**Review by vlica-harness — DISMISSED** [review]

**Comment by namantalaycha10** [comment]
trigger ut5

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasStepHelper.java [hunk 2]
```diff
@@ -2120,4 +2121,36 @@ private boolean isTasManifest(Map<String, Object> map) {
     }
     return false;
   }
+
+  private void validateArtifactBundleManifestStore(List<ManifestOutcome> orderedManifestOutcomes) {
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasStepHelper.java:2125]: does this validation, handle all the validations mentioned in tech spec, including override ones?

Comment (namantalaycha10) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasStepHelper.java:2125]: @RishabhGupta34 , those I will do with this ticket [https://harness.atlassian.net/browse/CDS-82975](https://harness.atlassian.net/browse/CDS-82975)

**Review by RishabhGupta34 — APPROVED** [review]

**Review by RishabhGupta34 — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by namantalaycha10** [comment]
trigger review

**Pull request merged**